plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -912,16 +912,26 @@ public VersionNumber getLatestPluginVersion(String pluginName) {
             String pluginName = dependency.getString("name");
             String pluginVersion = dependency.getString("version");
             Plugin dependentPlugin = new Plugin(pluginName, pluginVersion, null, null);
-            if (useLatestSpecified && plugin.isLatest() || useLatestAll) {
-                VersionNumber latestPluginVersion = getLatestPluginVersion(pluginName);
-                dependentPlugin.setVersion(latestPluginVersion);
-                dependentPlugin.setLatest(true);
-            }
             dependentPlugin.setOptional(dependency.getBoolean("optional"));
-
-            dependentPlugins.add(dependentPlugin);
             dependentPlugin.setParent(plugin);
-        }
+
+            try {
+                if (useLatestSpecified && plugin.isLatest() || useLatestAll) {
+                    VersionNumber latestPluginVersion = getLatestPluginVersion(pluginName);
+                    dependentPlugin.setVersion(latestPluginVersion);
+                    dependentPlugin.setLatest(true);
+                }
+                dependentPlugins.add(dependentPlugin);
+            } catch (PluginNotFoundException e) {
+                if (!dependentPlugin.getOptional()) {
+                    throw e;
+                }
+                logWarning(String.format(
```

Comment:
I think this should be under `logVerbose` the user hasn't done anything wrong here, the plugin needs to be updated to remove the dependency

```suggestion
                logVerbose(String.format(
```